distribution/src/resources/config-tool/templates/conf/axis2/axis2.xml.j2

```diff
@@ -0,0 +1,717 @@
+        {% for message_formatter in custom_message_formatters %}
+        <messageFormatter contentType="{{message_formatter.content_type}}"
+                          class="{{message_formatter.class}}"/>
+        {% endfor %}
+    </messageFormatters>
+
+    <!-- ================================================= -->
+    <!--                Message Builders                   -->
+    <!-- ================================================= -->
+
+    <!-- Following content type to builder mapping can be used to implement support for -->
+    <!-- different message formats in Axis2. These message formats are expected to be -->
+    <!-- resolved based on the content type. -->
+    <messageBuilders>
+        <messageBuilder contentType="application/xml"
+                        class="{{message_builders.application_xml}}"/>
+        <messageBuilder contentType="application/x-www-form-urlencoded"
+                        class="{{message_builders.form_urlencoded}}"/>
+        <messageBuilder contentType="multipart/form-data"
+                        class="{{message_builders.multipart_form_data}}"/>
+        <messageBuilder contentType="application/json"
+                        class="{{message_builders.application_json}}"/>
+        <messageBuilder contentType="application/json/badgerfish"
+                        class="{{message_builders.json_badgerfish}}"/>
+        <messageBuilder contentType="text/javascript"
+                        class="{{message_builders.text_javascript}}"/>
+        <messageBuilder contentType="text/plain"
+                        class="{{message_builders.text_plain}}"/>
+        <messageBuilder contentType="application/octet-stream"
+                        class="{{message_builders.octet_stream}}"/>
+        <messageBuilder contentType="application/binary"
+                        class="{{message_builders.application_binary}}"/>
+        {% for message_builder in custom_message_builders %}
+        <messageBuilder contentType="{{message_builder.content_type}}"
+                        class="{{message_builder.class}}"/>
+        {% endfor %}
+    </messageBuilders>
+
+    <!-- ================================================= -->
+    <!--             Transport Ins (Listeners)             -->
+    <!-- ================================================= -->
+
+    {% if transport.http.listener.enable == true %}
+    <transportReceiver name="http" class="{{transport.http.listener.class}}">
+        {% for parameter_key,parameter_value in transport.http.listener.parameter.items() %}
+        <parameter name="{{parameter_key}}" locked="false">{{parameter_value}}</parameter>
+        {% endfor %}
+    </transportReceiver>
+    {% endif %}
+
+    <transportReceiver name="https" class="{{transport.https.listener.class}}">
+        {% for parameter_key,parameter_value in transport.https.listener.parameter.items() %}
+        <parameter name="{{parameter_key}}" locked="false">{{parameter_value}}</parameter>
```

Comment:
missing <parameter name="non-blocking" locked="false">true</parameter>
for both http and https